src/agentscope/models/zhipu_model.py

```diff
@@ -0,0 +1,348 @@
+# -*- coding: utf-8 -*-
+"""Model wrapper for ZhipuAI models"""
+from abc import ABC
+from typing import Union, Any, List, Sequence
+
+from loguru import logger
+
+from .model import ModelWrapperBase, ModelResponse
+from ..message import MessageBase
+from ..utils.tools import _convert_to_str
+
+try:
+    import zhipuai
+except ImportError:
+    zhipuai = None
+
+
+class ZhipuAIWrapperBase(ModelWrapperBase, ABC):
+    """The model wrapper for ZhipuAI API."""
+
+    def __init__(
+        self,
+        config_name: str,
+        model_name: str = None,
+        api_key: str = None,
+        generate_args: dict = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize the zhipuai client.
+
+        Args:
+            config_name (`str`):
+                The name of the model config.
+            model_name (`str`, default `None`):
+                The name of the model to use in ZhipuAI API.
+            api_key (`str`, default `None`):
+                The API key for ZhipuAI API. If not specified, it will
+                be read from the environment variable.
+            generate_args (`dict`, default `None`):
+                The extra keyword arguments used in zhipuai api generation,
+                e.g. `temperature`, `seed`.
+        """
+
+        if model_name is None:
+            model_name = config_name
+            logger.warning("model_name is not set, use config_name instead.")
+
+        super().__init__(config_name=config_name)
+
+        if zhipuai is None:
+            raise ImportError(
+                "Cannot find zhipuai package in current python environment.",
+            )
+
+        self.model_name = model_name
+        self.generate_args = generate_args or {}
+
+        self.client = zhipuai.ZhipuAI(
```

Comment:
Do we need to consider other arguments here to initialize a client for ZhipuAI API?

Reply:
Added **kwargs to clients.

Reply:
In the ZhipuAi's examples, they only use the api_key, but actually arguments such as `timeout` can be set.